jobs/periodic/sync-to-webflow/sync-courses.ts

```diff
@@ -0,0 +1,172 @@
+import moment, { Moment } from 'moment';
+
+// This is needed so that the weekday will be translated properly.
+moment.locale('de');
+
+const collectionId = process.env.WEBFLOW_COLLECTION_ID;
+
+interface CourseDTO extends WebflowMetadata {
+    description: string;
+    instructor: string;
+    aboutinstructor: string;
+    subheading: string;
+
+    startingdate: string;
+    weekday: string;
+    courseduration: string; // like "45 min"
+    lessoncount: number;
+    time: string; // like 16:00 Uhr
+    appointments: string;
+
+    category: string;
+    link: string;
+    maxparticipants: number;
+    actualparticipants: number;
+    subject: string;
+
+    mingrade: number;
+    maxgrade: number;
+    priorknowledge: boolean;
+    agegroup: string;
+    level: string;
+
+    imagecopyrightinfo: string;
+    thumbnail: {
+        fileId: string; // TODO check if needed
+        url: string;
+        alt: string;
+    };
+    image: {
+        fileId: string; // TODO check if needed
+        url: string;
+        alt: string;
+    };
+}
+
+function courseDTOFactory(data: any): WebflowMetadata {
+    // TODO: has to be implemented
+    return data;
+}
+
+function generateHost(instructors: SubcourseInstructorsStudent[]): string {
+    const names = instructors.map((instructor) => `${instructor.student.firstname} ${instructor.student.lastname}`);
+    return names.join(', ');
+}
+
+function getStartDate(subcourse: Subcourse): Moment | null {
+    let earliestDate: Moment | null = null;
+    for (const lecture of subcourse.lecture) {
+        const startDate = moment(lecture.start);
+        if (earliestDate == null || startDate.isBefore(earliestDate)) {
+            earliestDate = moment(lecture.start);
+        }
+    }
+    return earliestDate;
+}
+
+function getCouseDuration(subcourse: Subcourse): number {
+    let duration = 0;
+    for (const lecture of subcourse.lecture) {
+        duration += lecture.duration;
+    }
+    return duration;
+}
+
+function listAppointments(subcourse: Subcourse): string {
+    let appointments = [];
+    for (const lecture of subcourse.lecture) {
+        const startDate = moment(lecture.start);
+        appointments.push(startDate.format('dddd, DD. MMMM YYYY, HH:mm [Uhr]'));
+    }
+    return appointments.join('\n');
+}
+
+function courseToDTO(subcourse: Subcourse): CourseDTO {
+    const startDate: Moment = getStartDate(subcourse) || moment();
+    const courseDTO: CourseDTO = {
+        ...emptyMetadata,
+
+        name: subcourse.course.name,
+        // TODO add databaseid field ot cms
+        databaseid: `${subcourse.id}`, // We are using a string to be safe for any case.
+
+        description: subcourse.course.description,
+        instructor: generateHost(subcourse.subcourse_instructors_student),
+        aboutinstructor: '', // TODO
+        subheading: subcourse.course.outline,
```

Comment:
The outline is currently not maintained, users cannot set it during course creation ...